crypto/property/property.c

```diff
@@ -440,10 +449,14 @@ int ossl_method_store_merge_global_properties(OSSL_METHOD_STORE *store,
 }
 
 
-static void impl_cache_flush_alg(ossl_uintmax_t idx, ALGORITHM *alg)
+static void impl_cache_flush_alg(ossl_uintmax_t idx, ALGORITHM *alg, void *arg)
 {
+    SPARSE_ARRAY_OF(ALGORITHM) *algs = arg;
+
     lh_QUERY_doall(alg->cache, &impl_cache_free);
     lh_QUERY_flush(alg->cache);
+    if (algs != NULL)
+        ossl_sa_ALGORITHM_set(algs, idx, NULL);
```

Comment:
Does this leak the hash table?

`lh_QUERY_free(alg->cache)` instead of `lh_QUERY_flush(alg->cache)` ??

Reply:
So you mean:
if (algs != NULL) {
    lh_QUERY_free(alg->cache);
    ossl_sa_ALGORITHM_set(algs, idx, NULL);
} else {
    lh_QUERY_flush(alg->cache);
}

Reply:
Yeah.

Reply:
It needed to do a bit more to cleanup than this as this was the only pointer to the algorithm.